iup/__init__.py

```diff
@@ -147,50 +147,75 @@ def to_incident(self) -> IncidentUptakeData:
         return out
 
 
-def get_nis(
-    data_path,
-    region_col,
-    date_col,
-    estimate_col,
-    rollout,
+def parse_nis(
+    path: str,
+    region_col: str,
+    date_col: str,
+    estimate_col: str,
+    date_format: str,
+    rollout: dt.date,
     filters=None,
 ) -> CumulativeUptakeData:
-    out = (
-        (pl.read_csv(data_path))
-        .with_columns(estimate=pl.col(estimate_col).cast(pl.Float64, strict=False))
-        .drop_nulls(subset=["estimate"])
-    )
+    frame = fetch_nis(path)
 
     if filters is not None:
-        filter_expr = pl.lit(True)
-        for k, v in filters.items():
-            filter_expr &= pl.col(k).is_in(pl.lit(v))
-        out = out.filter(filter_expr)
+        frame = apply_filters(frame, filters)
+
+    frame = select_columns(frame, region_col, date_col, estimate_col, date_format)
```

Comment:
optionally you could chain with https://docs.pola.rs/api/python/stable/reference/dataframe/api/polars.DataFrame.pipe.html